### PR TITLE
Added vault_utils.py

### PR DESCRIPTION
**Pull request by oharan2**

### CODE REVIEW

**Comment by oharan2** [comment]
!verified

File: ocp_utilities/vault_utils.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+LOGGER = logging.getLogger(__name__)
+
+
+def get_vault_config(vault_url, data_path, role_id, secret_id, verify=True):
+    """
+    Fetch vault data config
+    Args:
+        vault_url (str): vault url host to create client
+        data_path (str): path within vault to read from 'apps/data/..
```
Comment (rnetser) [inline ocp_utilities/vault_utils.py:17]: `'apps/data/..` - is it always this path? if not, please remove

File: ocp_utilities/vault_utils.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        role_id (str): for example: 'mps-qe'
```
Comment (rnetser) [inline ocp_utilities/vault_utils.py:18]: please update to explain what this is (based on vault) and remove any internal examples. this repo is public

File: ocp_utilities/vault_utils.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        secret_id (str): could be exported env variable
```
Comment (rnetser) [inline ocp_utilities/vault_utils.py:19]: please update the description to explain what is the secret id; should not include `could be exported env variable`; this is not relevant to this function

File: ocp_utilities/vault_utils.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        )
+
+        vault_client.auth.approle.login(
+            role_id=role_id,
+            secret_id=secret_id,
+        )
+
+        return vault_client.read(path=data_path)
+    except requests.exceptions.ConnectionError:
+        LOGGER.error("Failed to connect to vault.")
```
Comment (rnetser) [inline ocp_utilities/vault_utils.py:36]: please add exception text fo the log

**Review by rnetser — CHANGES_REQUESTED** [review]

**Comment by myakove** [comment]
I don't think this belongs here, this is OpenShift utilities repo

**Comment by rnetser** [comment]
@oharan2 as we no longer need this in multiple place, please move to the `integrations` repo

**Pull request closed without merging**